### PR TITLE
sets benchmark 

### PR DESCRIPTION
**Pull request by professorcode1**
Hey. 
I need your help. Please review my code and let me know what I am doing wrong. Its causing a segfault and I cannot figure out why.

### CODE REVIEW

**Comment by professorcode1** [comment]
@ntamas  please take a look

**Comment by szhorvat** [comment]
Can you please run it with AddressSanitizer enabled, and if you are still not sure what's wrong, post the AddressSanitizer output here?

Run `ccmake .` in the build directory to get the CMake GUI, then make these adjustments:

`CMAKE_BUILD_TYPE` should be `Debug`

`USE_SANITIZER` should be set to `Address`

Hit `c`, then `g`, then build and run the benchmark.

**Comment by professorcode1** [comment]
ok thanks. I will try that

**Comment by professorcode1** [comment]
@szhorvat please let me know how to add rule for a benchmark. Cause ` cmake --build . --target benchmark_igraph_set` is not working.

**Comment by GroteGnoom** [comment]
Did you add it to tests/CMakeLists.txt? If you search for `add_benchmarks` there, you'll see a list of benchmarks.

**Comment by ntamas** [comment]
@professorcode1 See the comment from @GroteGnoom ; you need to add the benchmark explicitly to `tests/CMakeLists.txt`, otherwise it won't compile and won't run. That's one thing. The other thing is that there's a failure now in the `igraph_weighted_cliques()` test case, but that's okay; the bug is tracked in https://github.com/igraph/igraph/issues/2306 and the test case is there to reproduce the bug. So it will keep on failing until we fix #2306, which I can start doing next week -- but it's completely unrelated to your code.

File: tests/benchmarks/igraph_set.c [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    igraph_integer_t induced_subgraph_vertice_count, igraph_integer_t benchmark_count,
+    const char* bench_message
+) {
+    igraph_vector_int_t edges;
+    printf("bench called\n");
+
+    ;
+    BENCH(bench_message,
+          REPEAT(
+                igraph_induced_subgraph_edges(graph, igraph_vss_range(0, induced_subgraph_vertice_count),&edges),
```
Comment (szhorvat) [inline tests/benchmarks/igraph_set.c:19]: You are using `edges` without initializing it first.

Comment (professorcode1) [reply tests/benchmarks/igraph_set.c]: Can't believe I still made that mistake. Thank you

**Comment by professorcode1** [comment]
@ntamas @GroteGnoom Thank you its working now.

**Comment by ntamas** [comment]
Thanks, looks good now! I'll merge this now and we can add other graph types to the benchmark later if needed.

**Pull request merged**